src/main/java/software/aws/neptune/NeptuneDriver.java

```diff
@@ -76,62 +68,75 @@ public boolean acceptsURL(final String url) throws SQLException {
-            int connectionTimeoutSeconds = connectionProperties.getConnectionTimeoutMillis() / 1000;
-            if (connectionTimeoutSeconds <= 0) {
-                connectionTimeoutSeconds = 1;
-            }
-
-            if (connection.isValid(connectionTimeoutSeconds)) {
-                return connection;
-            }
+            final DriverConnectionFactory connectionFactory = new DriverConnectionFactory();
+            return connectionFactory.getConnection(url, info);
         } catch (final Exception e) {
             if (e instanceof SQLException) {
                 throw (SQLException) e;
             }
             LOGGER.error("Unexpected error while creating connection:", e);
             return null;
         }
-        throw new SQLException("Error, failed to connect to database. Check endpoint, port, and configuration.");
     }
 
-    private ConnectionProperties connectionProperties(final String language, final Properties properties)
-            throws SQLException {
-        // TODO AN-550: Switch to map with class that holds Conn properties, key, query executor, etc.
-        if (OPENCYPHER.equalsIgnoreCase(language)) {
-            return new OpenCypherConnectionProperties(properties);
-        }
-        if (GREMLIN.equalsIgnoreCase(language) || SQL_GREMLIN.equalsIgnoreCase(language)) {
-            return new GremlinConnectionProperties(properties);
-        }
-        if (SPARQL.equalsIgnoreCase(language)) {
-            return new SparqlConnectionProperties(properties);
-        }
-        return new OpenCypherConnectionProperties(properties);
-    }
+    private class DriverConnectionFactory {
+        private final Pattern connStringPattern = Pattern.compile(CONN_STRING_PREFIX + "(\\w+)://(.*)");
+        private static final String OPENCYPHER = "opencypher";
+        private static final String GREMLIN = "gremlin";
+        private static final String SQL_GREMLIN = "sqlgremlin";
+        private static final String SPARQL = "sparql";
 
-    private String firstPropertyKey(final String language) {
-        // TODO AN-550: Switch to map with class that holds Conn properties, key, query executor, etc.
-        if (OPENCYPHER.equalsIgnoreCase(language)) {
-            return OpenCypherConnectionProperties.ENDPOINT_KEY;
+        public java.sql.Connection getConnection(final String url, final Properties info) throws SQLException {
+            final String language = getLanguage(url, connStringPattern);
+            final String propertyString = getPropertyString(url, connStringPattern);
+            final String firstPropertyKey = getFirstPropertyKey(language);
+            final Properties properties = setProperties(propertyString, firstPropertyKey, info);
+
+            if (OPENCYPHER.equalsIgnoreCase(language)) {
+                return new OpenCypherConnection(new OpenCypherConnectionProperties(properties));
+            }
+            if (GREMLIN.equalsIgnoreCase(language)) {
```

Comment:
A `switch` here might be more effective.

Reply:
I'll refactor into a `switch`